src/main/java/org/synyx/urlaubsverwaltung/person/PersonService.java

```diff
@@ -76,62 +71,68 @@ Person update(Integer id, String username, String lastName, String firstName, St
     /**
      * finds a {@link Person} in the database by its primary key.
      *
-     * @param  id  Integer the id of the person
-     *
-     * @return  optional {@link Person} for the given id
+     * @param id Integer the id of the person
+     * @return optional {@link Person} for the given id
      */
     Optional<Person> getPersonByID(Integer id);
 
 
     /**
      * finds a {@link Person} in the database by username.
      *
-     * @param  username  of the person
-     *
-     * @return  optional {@link Person} for the given username
+     * @param username of the person
+     * @return optional {@link Person} for the given username
      */
     Optional<Person> getPersonByUsername(String username);
 
 
     /**
      * finds all {@link Person}s in the database that have the given {@link Role}.
      *
-     * @param  role  {@link Role}
-     *
-     * @return  {@link List} of {@link Person}
+     * @param role {@link Role}
+     * @return {@link List} of {@link Person}
      */
     List<Person> getActivePersonsByRole(Role role);
 
 
     /**
      * returns all {@link Person}s that have the given {@link MailNotification} type.
      *
-     * @param  notification  by which the persons are filtered
-     *
-     * @return  list of persons with the given notification type
+     * @param notification by which the persons are filtered
+     * @return list of persons with the given notification type
      */
     List<Person> getPersonsWithNotificationType(MailNotification notification);
 
 
     /**
      * returns all active persons ordered by first name.
      *
-     * @return  returns all active persons
+     * @return returns all active persons
      */
     List<Person> getActivePersons();
 
 
     /**
      * returns all inactive persons ordered by first name.
      *
-     * @return  returns all inactive persons
+     * @return returns all inactive persons
      */
     List<Person> getInactivePersons();
 
     /**
      * This method allows to get the signed in user.
      *
-     * @return  user that is signed in
+     * @return user that is signed in
      */
     Person getSignedInUser();
+
+    /**
+     * Adds {@link Role#OFFICE} to the roles of the given person if no
+     * other active user with a office role is defined.
+     *
+     * @param person that maybe gets the role {@link Role#OFFICE}
+     * @return saved {@link Person} with {@link Role#OFFICE} rights
+     * if no other active person with {@link Role#OFFICE} is available.
+     */
+    Person appointAsOfficeUserIfNoOfficeUserPresent(Person person);
```

Comment:
Ich würde diese Methode gerne irgendwie verstecken und irgendwann durch einen Wizard o.ä. austauschen. Da durch diese Methode die chance besteht unglücklicherweise an eine Rolle zu gelangen, die der Benutzer nicht haben sollte. Wäre es besser diese Methode in eine separate Klasse im package `Security` aps package protected zu packen?

Reply:
Verstehe ich und das Ersetzen durch Wizzard etc. wäre optimal. 
Aber, nur weil man es verschleiert heißt es ja nicht, dass die Methode nicht da ist. Im Gegenzug erstellt man eine neue Klasse und zieht sich eine weitere Dependency rein? Was die Komplexität erhöht? Dachte, dass die Auflösung des PersonSyncServices gerade auch das zum Grund hatte?
Vielleicht eher ab V3 explizit als Deprecated markieren und den Wizzard auf die Roadmap setzen?

Reply:
Ja das könnte man machen. Direkt in v3 als deprecated und dann den Wizard auf die roadmap.

Reply:
Es ging mir eher darum die methode im security package package private zu lassen, da diese doch nur da gebraucht wird, sodass niemand aus anderen Paketen diese Methode verwendet.